### PR TITLE
team-list: establish java team

### PR DESCRIPTION
**Pull request by tomodachi94**
As discussed in #jvm:nixos.org on Matrix, the maintainers of the Java ecosystem in Nixpkgs feel that a team for Java would be helpful.

## Blockers
- [x] Waiting on responses from the following contributors to the Java Nixpkgs ecosystem (non-blocking after 2024-11-08):
    - [x] @chayleaf (said yes)
    - [x] @Infinidoge (said yes)
    - [x] @\\\\thiagokokada (said no)
    - [x] @\\\\emilazy (said no *for now*)
    - [x] @FliegendeWurst (said yes)
    - [x] @NeQuissimus (said no)
    - [ ] @shlevy (no GitHub activity in the entire month of October, probably won't respond)
    - [ ] @edwtjo  (no GitHub activity since 2024-09-27, not likely to respond)
    - [ ] @taku0 (active on GitHub)
- [x] Waiting on organization owners to add @NixOS\\\/java to the Nixpkgs repository (done by lassulus)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tomodachi94** [comment]
@shlevy @NeQuissimus I saw that you are two very prolific contributors in the Nixpkgs Java ecosystem, but I couldn't find you on Matrix. Are either of you interested in joining the team?

**Comment by Infinidoge** [comment]
Should the Java team be added to the JDK packages themselves?

**Comment by tomodachi94** [comment]
Done. It appears to work as expected.

**Comment by tomodachi94** [comment]
@edwtjo @taku0 are either of you interested as well? I noticed you both were maintainers of some packages.

File: ci/OWNERS [hunk 1]
```diff
@@ -206,6 +206,11 @@ pkgs/data/misc/cacert/ @ajs124 @lukegb @mweinelt
 pkgs/development/libraries/nss/ @ajs124 @lukegb @mweinelt
 pkgs/development/python-modules/buildcatrust/ @ajs124 @lukegb @mweinelt
 
+# Java
+/doc/languages-frameworks/java.section.md @NixOS/java
```
Comment (chayleaf) [inline ci/OWNERS:210]:
```suggestion
/doc/languages-frameworks/java.section.md @NixOS/java
/doc/languages-frameworks/gradle.section.md @NixOS/java
```

Comment (tomodachi94) [reply ci/OWNERS:210]: Good catch, thank you :heart:

**Comment by tomodachi94** [comment]
If anyone is interested in joining this team (thusly helping with Java maintenance) and I didn't ping you, feel free to leave a comment here or message me on Matrix (`@tomodachi94:matrix.org`).

**Comment by tomodachi94** [comment]
Let's merge this on Friday or Saturday. A response by the people mentioned above before that time would be appreciated, but we can always add team members later.

**Comment by tomodachi94** [comment]
@FliegendeWurst are you interested as well? I noticed you were active on several Java-related tickets.

**Comment by tomodachi94** [comment]
@vcunat are you interested as well? You've done some work on Java things.

**Comment by vcunat** [comment]
I'm not aware.  I certainly don't feel like a Java person at all.

**Comment by tomodachi94** [comment]
As for why I reached out to you (and most others on the list in the description), you had 30 commits in Java-adjacent files according to this estimate from about 5 days ago: https://gist.github.com/tomodachi94/37564d70235ee290966207948dcdb69d

**Comment by NeQuissimus** [comment]
While I would be interested, I don't think I have time right now

**Comment by tomodachi94** [comment]
Responses from the above-mentioned people who haven't responded are now non-blocking, since it's been a week. We can always add them to the team in a later PR.

**Review by FliegendeWurst — APPROVED** [review]

**Review by Infinidoge — APPROVED** [review]

**Review by chayleaf — APPROVED** [review]

**Pull request merged**